/home/travis/dpdk/lib/librte_eal/linuxapp/eal/eal_memory.c:28:18: fatal error: numa.h: No such file or directory
4271compilation terminated.

I think we need to add libnuma-dev

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,7 +3,8 @@ addons:
   apt:
     packages:
     - libncurses-dev
-    - linux-headers-3.13.0-40-generic
+    - linux-headers-4.15.0-58-generic
+    - libnuma-dev
 
 cache:
   directories:
@@ -13,9 +14,9 @@ before_install:
   - >
     [ -z "$(ls -A $HOME/dpdk)" ] &&
     cd $HOME/dpdk &&
-    wget http://fast.dpdk.org/rel/dpdk-16.11.1.tar.xz &&
-    tar --strip-components=1 -axvf dpdk-16.11.1.tar.xz &&
-    export RTE_KERNELDIR=/lib/modules/3.13.0-40-generic/build &&
+    wget http://fast.dpdk.org/rel/dpdk-18.11.5.tar.xz &&
+    tar --strip-components=1 -axvf dpdk-18.11.5.tar.xz &&
+    export RTE_KERNELDIR=/lib/modules/4.15.0-58-generic/build &&
     make install T=x86_64-native-linuxapp-gcc; true
 
 script:

--- a/src/core_capture.c
+++ b/src/core_capture.c
@@ -5,6 +5,7 @@
 #include <rte_log.h>
 #include <rte_lcore.h>
 #include <rte_ethdev.h>
+#include <rte_version.h>
 
 #include "core_capture.h"
 
@@ -17,6 +18,7 @@ int capture_core(const struct core_capture_config * config) {
   struct rte_mbuf *bufs[DPDKCAP_CAPTURE_BURST_SIZE];
   uint16_t nb_rx;
   int nb_rx_enqueued;
+  int i;
 
   RTE_LOG(INFO, DPDKCAP, "Core %u is capturing packets for port %u\n",
       rte_lcore_id(), config->port);
@@ -39,17 +41,22 @@ int capture_core(const struct core_capture_config * config) {
     nb_rx = rte_eth_rx_burst(config->port, config->queue,
         bufs, DPDKCAP_CAPTURE_BURST_SIZE);
     if (likely(nb_rx > 0)) {
-      nb_rx_enqueued = rte_ring_enqueue_bulk(config->ring, (void*) bufs,
+#if RTE_VERSION >= RTE_VERSION_NUM(17,5,0,16)
+      nb_rx_enqueued = rte_ring_enqueue_burst(config->ring, (void*) bufs,
+          nb_rx, NULL);
+#else
+      nb_rx_enqueued = rte_ring_enqueue_burst(config->ring, (void*) bufs,
           nb_rx);
+#endif
 
       /* Update stats */
-      if(nb_rx_enqueued == 0 || nb_rx_enqueued == -EDQUOT) {
+      if(nb_rx_enqueued == nb_rx) {
         config->stats->packets+=nb_rx;
       } else {
         config->stats->missed_packets+=nb_rx;
         /* Free whatever we can't put in the write ring */
-        for (nb_rx_enqueued=0; nb_rx_enqueued < nb_rx; nb_rx_enqueued++) {
-          rte_pktmbuf_free(bufs[nb_rx_enqueued]);
+        for (i=nb_rx_enqueued; i < nb_rx; i++) {
+          rte_pktmbuf_free(bufs[i]);
         }
       }
     }

--- a/src/core_write.c
+++ b/src/core_write.c
@@ -10,6 +10,7 @@
 #include <rte_log.h>
 #include <rte_mbuf.h>
 #include <rte_branch_prediction.h>
+#include <rte_version.h>
 
 #include "lzo/lzowrite.h"
 #include "pcap.h"
@@ -227,14 +228,13 @@ int write_core(const struct core_write_config * config) {
     }
 
     //Get packets from the ring
-    to_write = rte_ring_dequeue_bulk(config->ring, (void*) dequeued,
-        DPDKCAP_WRITE_BURST_SIZE);
-    if (likely(to_write==0)) {
-      to_write = DPDKCAP_WRITE_BURST_SIZE;
-    } else {
+#if RTE_VERSION >= RTE_VERSION_NUM(17,5,0,16)
+      to_write = rte_ring_dequeue_burst(config->ring, (void*)dequeued,
+          DPDKCAP_WRITE_BURST_SIZE, NULL);
+#else
       to_write = rte_ring_dequeue_burst(config->ring, (void*)dequeued,
           DPDKCAP_WRITE_BURST_SIZE);
-    }
+#endif
 
     //Update stats
     config->stats->packets += to_write;

--- a/src/dpdkcap.c
+++ b/src/dpdkcap.c
@@ -10,6 +10,7 @@
 #include <rte_ethdev.h>
 #include <rte_errno.h>
 #include <rte_string_fns.h>
+#include <rte_version.h>
 
 #include "pcap.h"
 #include "core_write.h"
@@ -272,11 +273,18 @@ static int port_init(
   struct rte_eth_dev_info dev_info;
   int retval;
   uint16_t q;
+  uint16_t dev_count;
 
   /* Check if the port id is valid */
+#if RTE_VERSION >= RTE_VERSION_NUM(18,11,3,16)
+  dev_count = rte_eth_dev_count_avail()-1;
+#else
+  dev_count = rte_eth_dev_count()-1;
+#endif
+
   if(rte_eth_dev_is_valid_port(port)==0) {
-    RTE_LOG(ERR, DPDKCAP, "Port identifier %d out of range (0 to %d) or not"\
-       " attached.\n", port, rte_eth_dev_count()-1);
+     RTE_LOG(ERR, DPDKCAP, "Port identifier %d out of range (0 to %d) or not"\
+       " attached.\n", port, dev_count);
     return -EINVAL;
   }
 
@@ -383,6 +391,7 @@ int main(int argc, char *argv[]) {
   unsigned int required_cores;
   unsigned int core_index;
   int result;
+  uint16_t dev_count;
   FILE * log_file;
 
   /* Initialize the Environment Abstraction Layer (EAL). */
@@ -413,8 +422,12 @@ int main(int argc, char *argv[]) {
   argp_parse(&argp, argc, argv, 0, 0, &arguments);
 
   /* Set log level */
+#if RTE_VERSION >= RTE_VERSION_NUM(17,5,0,16)
+  rte_log_set_level(RTE_LOG_DEBUG, RTE_LOG_DEBUG);
+#else
   rte_set_log_type(RTE_LOGTYPE_DPDKCAP, 1);
   rte_set_log_level(RTE_LOG_DEBUG);
+#endif
 
   /* Change log stream if needed */
   if(arguments.log_file) {
@@ -447,15 +460,21 @@ int main(int argc, char *argv[]) {
     strcat(arguments.output_file_template, ".lzo");
 
   /* Check if at least one port is available */
-  if (rte_eth_dev_count() == 0)
+#if RTE_VERSION >= RTE_VERSION_NUM(18,11,3,16)
+  dev_count=rte_eth_dev_count_avail();
+#else
+  dev_count=rte_eth_dev_count();
+#endif
+
+  if (dev_count == 0)
     rte_exit(EXIT_FAILURE, "Error: No port available.\n");
 
   /* Fills in the number of rx descriptors matrix */
-  unsigned long * num_rx_desc_matrix = calloc(rte_eth_dev_count(), sizeof(int));
+  unsigned long * num_rx_desc_matrix = calloc(dev_count, sizeof(int));
   if (arguments.num_rx_desc_str_matrix != NULL &&
-      parse_matrix_opt(arguments.num_rx_desc_str_matrix,
-        num_rx_desc_matrix, rte_eth_dev_count()) < 0) {
-    rte_exit(EXIT_FAILURE, "Invalid RX descriptors matrix.\n");
+      parse_matrix_opt(arguments.num_rx_desc_str_matrix,                        
+        num_rx_desc_matrix, dev_count) < 0) {
+    rte_exit(EXIT_FAILURE, "Invalid RX descriptors matrix.\n");                 
   }
 
   /* Creates the port list */
@@ -463,7 +482,7 @@ int main(int argc, char *argv[]) {
   for (i = 0; i < 64; i++) {
     if (! ((uint64_t)(1ULL << i) & arguments.portmask))
       continue;
-    if (i<rte_eth_dev_count())
+    if (i<dev_count)
       portlist[nb_ports++] = i;
     else
       RTE_LOG(WARNING, DPDKCAP, "Warning: port %d is in portmask, " \

